### ISSUE TITLE
Roadmap

### ISSUE THREAD

**Issue by jpdillingham** [slskd/slskd#413]
# [0.16: Improved Search Experience](https://github.com/slskd/slskd/milestone/2)

The current search experience restricts users to one search at a time, doesn't display results until the search is complete, and doesn't give the user any control over when a search is determined to have been completed.

This milestone is dedicated to a ground-up refactor of the search functionality to resolve these shortcomings, both to improve overall user experience and to enable search automation, for which concurrent searches are critical.

# [0.17: Guided Search](https://github.com/slskd/slskd/milestone/3)

Presently, users enter a simple search phrase and must review results to determine whether they are suitable. For example, when searching for a full album, the user must know the track listing ahead of time, and must compare results against the listing while looking for missing files, etc.

The Guided Search feature will allow users to select an artist and an album to search for, and will use data from MusicBrainz (and potentially Discogs) to rank and automatically select the best results as determined by user-defined criteria.

# [0.18: Post-Download Automation](https://github.com/slskd/slskd/milestone/4)

When downloads finish, users may want the option to perform various tasks; moving files between network shares, setting file permissions, post-processing metadata with a program like [beets](https://beets.io/), or notifying external applications.

The Post-Download Automation milestone will introduce a suite of tools and options to give users the ability to accomplish these tasks, and more.

**Comment by jpdillingham** [comment]
I've removed the min-milestone 0.16.x (misc cleanup and quality of life improvements) as the biggest chunks of that are now complete.

Based on feedback I've gotten from the community, I've moved the Guided Search milestone from 0.17 to 0.19 and introduced the new 0.17 milestone Share Scanning at Scale

Effectively this pushes Lidarr integration out again, but in the spirit of Agile software development I need to work on the most important things first.

**Comment by jpdillingham** [comment]
I've just updated the roadmap again.  Changes:

* Added the 1.00 milestone
* Removed numbering from Post-Download Automation and Guided Search milestones

Currently the versioning scheme (aligned with numeric milestones) only makes sense to me, and I don't even get much benefit out of it, so I want to move to semver.  I had always planned to start using semver after I had gotten to 1.00, and I believe the app is near or perhaps past 1.00 at this point.

This pushes download automation out again, and again it is because people are asking for things other than download automation.  This does remain a personal goal, so it will be done eventually even if it is out-of-band.

**Comment by aopom** [comment]
Many thanks for this transparency, it's super cool to know where the project is going! Post-Download Automation will be a great feature to have. Exciting!

**Comment by KINGLIFER** [comment]
Soooo this is a thing?
any way to Bata test?

**Comment by jpdillingham** [comment]
@KINGLIFER I updated the Lidarr issue (#451):

> Since Lidarr integration was recently brought up, I want to be transparent; my day jobs have been demanding nearly all of my time for the past 6 months or so and that will likely continue into the foreseeable future.
> 
> I am committed to continuing work on slskd and eventually attempting to deliver this integration, but right now I am slow and even simple things take a lot of time to ship. Due to the number of things that need to be in place for this work to begin (see #413) the likelihood of this happening in 2024 is next to zero.
> 
> That being said, I've added the `help wanted` tag and would encourage anyone that knows or would like to learn C#/.NET to reach out if they are serious about helping.

Unfortunately the issues in the [QOL](https://github.com/slskd/slskd/milestone/6) milestone are growing faster than I can resolve them, but I am hopeful that I can set some of those aside to work on some of the net new features that will move slskd closer to automation (this was always the goal, just taking a long time to get there).

**Comment by allquiet-hub** [comment]
Hello. I have some knowledge with C#, i might be able to start working on this integration, since it would also benefit my own usage of both apps. If you can point me in the right direction, like in what project to build the integration and so, to kickstart the development.

**Comment by allquiet-hub** [comment]
What do you say, @jpdillingham ? Sorry to ping you.

**Comment by jpdillingham** [comment]
@allquiet-hub thanks for volunteering! And sorry for the delayed response.  I'll add some thoughts to #451

**Comment by Chris230291** [comment]
I have patiently been waiting for Lidarr integration for years now.
Maybe it's time to abandon the idea and just build an automatic downloading and tagging system into Slskd?
I don't think anyone is invested in Lidarr specifically?
Perhaps use Discogs (which is better) rather than MusicBrainz?

**Comment by zGenMedia** [comment]
One day some one rich will just take over these dormant goldmines. Until then. We suffer.

**Comment by super-cooper** [comment]
@Chris230291 While we wait, one thing that I've heard works is setting up a "Usenet Black Hole" download client in Lidarr which points at your slskd download directory. It doesn't seem perfect and requires a little more work than doing everything through Lidarr, but imo doesn't seem too bad. Worth giving a try.

**Comment by Chris230291** [comment]
> @Chris230291 While we wait, one thing that I've heard works is setting up a "Usenet Black Hole" download client in Lidarr which points at your slskd download directory. It doesn't seem perfect and requires a little more work than doing everything through Lidarr, but imo doesn't seem too bad. Worth giving a try.

Yes I know. It's not automated, though, which is what we all want.

**Comment by SawkeeReemo** [comment]
For those looking for Lidarr integration, I just found Soularr tonight that’s built to work with slams. It works by auto-searching for anything you have monitored. I decided to set up a separate instance of Lidarr to run with this so I can control what I use slskd to search for. But check it out, it’s pretty cool: https://soularr.net/

**Comment by allquiet-hub** [comment]
Quick addition, in the end I did build the integration for Lidarr.
Recently the guys over at Lidarr did heaps of work on the plugins function. You need to use Lidarr (plugins branch) version 2.9.4.4538 at a minimum. The plugin is available at https://github.com/allquiet-hub/Lidarr.Plugin.Slskd to be directly consumed by the Lidarr plugins manager. If you have issues or feedback use the issues section on the repository or hop over to the Lidarr discord, where I'm pretty active.

**Comment by zGenMedia** [comment]
Is this changing my set up? I legit donr have time to sit and guess how this new one is done. Seriously.

**Comment by Chris230291** [comment]
> Quick addition, in the end I did build the integration for Lidarr. Recently the guys over at Lidarr did heaps of work on the plugins function. You need to use Lidarr (plugins branch) version 2.9.4.4538 at a minimum. The plugin is available at https://github.com/allquiet-hub/Lidarr.Plugin.Slskd to be directly consumed by the Lidarr plugins manager. If you have issues or feedback use the issues section on the repository or hop over to the Lidarr discord, where I'm pretty active.

Awesome, thanks.
Is it a separate branch while the plugin system is being tested?
I assume it will merge into main if everything is OK?

**Comment by jpdillingham** [comment]
I've deleted a number of off-topic comments and will be closing and locking this as it has gone far beyond it's usefulness.
